src/sensors/bmi323sensor.h

```diff
@@ -0,0 +1,184 @@
+     */
+    void startCalibration(int calibrationType) override final;
+
+    static int8_t i2cRead(uint8_t registerAddress, uint8_t *registerData, uint32_t length, void *interfacePointer);
+    static int8_t i2cWrite(uint8_t registerAddress, const uint8_t *registerData, uint32_t length, void *interfacePointer);
+    static void delayUs(uint32_t period, void *interfacePointer);
+
+    uint8_t address;
+    SensorStatus m_status = SensorStatus::SENSOR_OFFLINE;
+
+private:
+    /**
+     * @brief Extracts one frame from the FIFO buffer
+     * 
+     * The FIFO buffer is an array of uint_8t divided into 16 bytes chunks (if everything is enabled)
+     * The first 3 words (6 bytes) are the accelerometer data (X, Y, Z)
+     * The next 3 words (6 bytes) are the gyroscope data (X, Y, Z)
+     * The next word (2 bytes) is the temperature data
+     * The last word (2 bytes) is the time data
+     * 
+     * The least significant byte is first
+     * The most significant byte is last
+     * 
+     * Each data can be a dummy data, in which case the value of the first word
+     * will be equal to BMI_FIFO_<SENSOR>_DUMMY_FRAME
+     * 
+     * All FIFO data buffer first two bytes are dummy bytes, they need to be skiped
+     * It is done by adding bmi323.dummy_byte to the index after reseting them
+     * 
+     * If one of the data is disabled then the buffer will be shorter by the data length
+     * In this case temperature is disabled so the buffer is 14 bytes long
+     * 
+     * Data example:
+     * [<ACCEL_X_LSB>, <ACCEL_X_MSB>, <ACCEL_Y_LSB>, <ACCEL_Y_MSB>, <ACCEL_Z_LSB>,
+     *  <ACCEL_Z_MSB>, <GYRO_X_LSB>, <GYRO_X_MSB>, <GYRO_Y_LSB>, <GYRO_Y_MSB>,
+     *  <GYRO_Z_LSB>, <GYRO_Z_MSB>, <TEMP_LSB>, <TEMP_MSB>, <TIME_LSB>, <TIME_MSB>]
+    */
+    uint8_t extractFrame(uint8_t *data, uint8_t index, float *accelData, float *gyroData, float *tempData);
+
+    /**
+     * @brief Print calibration data
+    */
+    void printCalibrationData();
+
+    /**
+     * @brief Apply calibration data from Magnometer
+    */
+    void applyMagCalibrationAndScale(float Mxyz[3]);
+
+    /**
+     * @brief Get a single temperature reading
+    */
+    float getTemperature();
+
+    /**
+     * @brief Saves temperature calibration data
+    */
+    void saveTemperatureCalibration() override final;
+
+    /**
+     * SlimeVR Stuff
+    */
+    SlimeVR::Sensors::SensorFusionRestDetect m_sfusion;
+    SlimeVR::Configuration::BMI323CalibrationConfig m_calibration;
+    SlimeVR::Configuration::Configuration m_configuration;
+
+    GyroTemperatureCalibrator* gyroTempCalibrator = nullptr;
+    float GOxyzStaticTempCompensated[3] = {0.0, 0.0, 0.0};
+
+    /**
+     * BMI323 FIFO reading
+    */
+    const uint8_t ACCEL_VALID = 0x01;   // Bit 0 represents accelerometer data validity
+    const uint8_t GYRO_VALID = 0x02;    // Bit 1 represents gyroscope data validity
+    const uint8_t TEMP_VALID = 0x04;    // Bit 2 represents temperature data validity
```

Comment:
Make these into `#define`s, `constexpr`s or `static`s, otherwise they get recreated with every instance of the sensor. It's not a lot of memory but still.